### PR TITLE
Set up DB connection, implement DB aspects of exile/unexile commands

### PR DESCRIPTION
**Pull request by KingPieson**
Set up DB connection on bot spin up. Create DB files to run methods, create exile and unexile records when commands are run

### CODE REVIEW

File: src/services/exile_service.py [hunk 1]
```diff
@@ -1,34 +1,81 @@
...
+        user, role_to_add=Role.EXILED, role_to_remove=Role.VERIFIED
     )
 
-    # Look up 
-
-    log_info_and_embed(logging_embed, logger, f"because {reason}")
+    # message user
+    await send_dm(
+        user, f"You are being exiled from NAUR FFXIV for the following reason: {reason}"
+    )
```
Comment (brtran4) [inline src/services/exile_service.py:27]: i think we should move this near the end after all the db interactions are completed

File: src/services/exile_service.py [hunk 2]
```diff
@@ -1,34 +1,81 @@
...
-    await add_and_remove_role(user, role_to_add=Role.EXILED, role_to_remove=Role.VERIFIED)
+    # add exile entry into DB
+    start_timestamp = datetime.datetime.now(datetime.timezone.utc)
+    end_timestamp = None
+    exile_status = ExileStatus.INDEFINITE_EXILE
+    if duration:
+        end_timestamp = start_timestamp + duration
+        exile_status = ExileStatus.TIMED_EXILED
 
+    exile = Exile(None, db_user.user_id, reason, exile_status.value, start_timestamp, end_timestamp)
```
Comment (brtran4) [inline src/services/exile_service.py:50]: is there a reason why the class will contain an exile_id if it's not actually holding that information?

File: src/services/exile_service.py [hunk 3]
```diff
@@ -1,34 +1,81 @@
...
+    logging_embed: discord.Embed,
+    user: discord.Member,
+    duration: Optional[datetime.timedelta],
+    reason: str,
 ):
-    # TODO implement this functionality
-    log_info_and_embed(
-        logging_embed, logger, f"going to exile {user.mention} for {duration}"
+    # change user role
+    await add_and_remove_role(
```
Comment (brtran4) [inline src/services/exile_service.py:20]: looking at the function definition i think we'll have a bug where the user could potentially not have the role, but we're still attempting to remove it. this is a hard constraint on exile where the user needs the verified role

**Review by brtran4 — CHANGES_REQUESTED** [review]

File: src/services/exile_service.py (hunk 3 above)
Comment (KingPieson) [reply src/services/exile_service.py:20]: I think I had just misread a request that Claire had about unexile, I agree we should move this to then end, updated

File: src/services/exile_service.py (hunk 1 above)
Comment (KingPieson) [reply src/services/exile_service.py:27]: I think I had just misread a request that Claire had about unexile, I agree we should move this to then end, updated

File: src/services/exile_service.py (hunk 3 above)
Comment (KingPieson) [reply src/services/exile_service.py]: Makes sense, I've added a check to the service and updated the command to better pass around errors

File: src/services/exile_service.py (hunk 2 above)
Comment (KingPieson) [reply src/services/exile_service.py:49]: I still want to think about if there's a better way to pass around data without bloating the number of classes, but the idea is that we want postgres to take care of generating the actual PK of the row, so it's set up as null when it's first being created

Comment (brtran4) [reply src/services/exile_service.py:49]: that makes sense, i'm just not sure if there's a situation where we will need to throw around the exile id around except for getting the list of exiles. it's not detailed in the spec right now so i'm not sure if we're actually going to be doing that

File: src/services/exile_service.py (hunk 1 above)
Comment (brtran4) [reply src/services/exile_service.py]: i think we deleted this line on accident

Comment (KingPieson) [reply src/services/exile_service.py]: That's just a merge conflict thing, there was two calls to each of those and now they exist at the bottom of the method

**Review by brtran4 — APPROVED** [review]

**Pull request merged**